terratorch/tasks/base_task.py

```diff
@@ -156,3 +160,16 @@ def _do_plot_samples(self, batch_index):
             and hasattr(self.logger, "experiment")
             and (hasattr(self.logger.experiment, "add_figure") or hasattr(self.logger.experiment, "log_figure"))
         )
+
+    def record_metrics(self, dataloader_idx, y_hat_hard, y):
+
+        if self.path_to_record_metrics:
+            # Recording the metrics
+            metrics_record_ = self.test_metrics[dataloader_idx](y_hat_hard, y)
+            metrics_record_dict = {k: float(np.array(v.detach().cpu())) for k,v in metrics_record_.items()} 
+            metrics_record_list = [{"Metric": k, "Value": v} for k, v in metrics_record_dict.items()]
+            metrics_record = pd.DataFrame(data=metrics_record_list)
+
+            filename = os.path.join(self.path_to_record_metrics, "metrics.csv")
+            metrics_record.to_csv(filename)
+
```

Comment:
@Joao-L-S-Almeida just to make the code safer I'd throw an exception in case path_to_record_metrics is empty (failfast)

Reply:
sorry @Joao-L-S-Almeida just realized that you are pushing the decision logic into the function, which is a different style I'm used to but I like it as the decision logic does not need to be duplicated at caller side and also reduces control flow, thanks!!!